File: ruby/lib/simple_inline_text_annotation/relation_validator.rb
# frozen_string_literal: true

class SimpleInlineTextAnnotation
  module RelationValidator
    def validate_relations(relations)
      remove_incomplete_key_relations(relations)
    end

    private

    def remove_incomplete_key_relations(relations)
      relations.select { |relation| relation["subj"] && relation["pred"] && relation["obj"] }
    end
  end
end

File: js/src/relation_validator.mjs
class RelationValidator {
  validateRelations(relations) {
    return this.#removeIncompleteKeyRelations(relations);
  }

  #removeIncompleteKeyRelations(relations) {
    return relations.filter((relation) => {
      return relation.subj && relation.pred && relation.obj;
    });
  }
}

export default RelationValidator;
